src/internal/m365/restore.go

```diff
@@ -12,71 +12,65 @@ import (
 	"github.com/alcionai/corso/src/internal/m365/onedrive"
 	"github.com/alcionai/corso/src/internal/m365/sharepoint"
 	"github.com/alcionai/corso/src/internal/m365/support"
+	"github.com/alcionai/corso/src/internal/operations/inject"
 	"github.com/alcionai/corso/src/pkg/backup/details"
-	"github.com/alcionai/corso/src/pkg/control"
 	"github.com/alcionai/corso/src/pkg/count"
 	"github.com/alcionai/corso/src/pkg/fault"
-	"github.com/alcionai/corso/src/pkg/selectors"
+	"github.com/alcionai/corso/src/pkg/path"
 )
 
 // ConsumeRestoreCollections restores data from the specified collections
 // into M365 using the GraphAPI.
 // SideEffect: status is updated at the completion of operation
 func (ctrl *Controller) ConsumeRestoreCollections(
 	ctx context.Context,
-	backupVersion int,
-	sels selectors.Selector,
-	restoreCfg control.RestoreConfig,
-	opts control.Options,
+	rcc inject.RestoreConsumerConfig,
 	dcs []data.RestoreCollection,
 	errs *fault.Bus,
 	ctr *count.Bus,
 ) (*details.Details, error) {
 	ctx, end := diagnostics.Span(ctx, "m365:restore")
 	defer end()
 
-	ctx = graph.BindRateLimiterConfig(ctx, graph.LimiterCfg{Service: sels.PathService()})
-	ctx = clues.Add(ctx, "restore_config", restoreCfg) // TODO(rkeepers): needs PII control
+	ctx = graph.BindRateLimiterConfig(ctx, graph.LimiterCfg{Service: rcc.Selector.PathService()})
+	ctx = clues.Add(ctx, "restore_config", rcc.RestoreConfig) // TODO(rkeepers): needs PII control
 
 	if len(dcs) == 0 {
 		return nil, clues.New("no data collections to restore")
 	}
 
 	var (
-		status *support.ControllerOperationStatus
-		deets  = &details.Builder{}
-		err    error
+		service = rcc.Selector.PathService()
+		status  *support.ControllerOperationStatus
+		deets   = &details.Builder{}
+		err     error
 	)
 
-	switch sels.Service {
-	case selectors.ServiceExchange:
-		status, err = exchange.ConsumeRestoreCollections(ctx, ctrl.AC, restoreCfg, dcs, deets, errs, ctr)
-	case selectors.ServiceOneDrive:
+	switch service {
+	case path.ExchangeService:
+		status, err = exchange.ConsumeRestoreCollections(ctx, ctrl.AC, rcc, dcs, deets, errs, ctr)
```

Comment:
To be consistent with other similar invocations.
```suggestion
		status, err = exchange.ConsumeRestoreCollections(
		ctx,
		ctrl.AC,
		rcc,
		dcs,
		deets,
		errs,
		ctr)
```